[Hardware/NVIDIA/Modelopt] Fix modelopt forward method for v1 torch.compile

pavanimajety opened the pull request:
This is a simple fix to wrap ModelWeightParameter class object back to Parameter class for making it torch.compile compatible. 

cc/for review: @LucasWilkinson @mgoin 


<!--- pyml disable-next-line no-emphasis-as-heading -->

pavanimajety committed:
Fix Parameter loading for NVFP4 Linear & MoE Layers for torch.compile

Signed-off-by: Pavani Majety <pmajety@nvidia.com>

pavanimajety committed:
Fix lint

Signed-off-by: Pavani Majety <pmajety@nvidia.com>

pavanimajety committed:
Make it less verbose

Signed-off-by: Pavani Majety <pmajety@nvidia.com>

LucasWilkinson commented on vllm/model_executor/layers/quantization/modelopt.py:
```diff
@@ -587,7 +584,6 @@ def swizzle_blockscale(self, scale: torch.tensor):
 
     def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
         # GEMM 1
-
```
nit: remove un-related format change

pavanimajety replied:
I believe precommit modified this.

LucasWilkinson commented on vllm/model_executor/layers/quantization/modelopt.py:
```diff
@@ -426,11 +427,7 @@ def apply(
-        # for input only the contracting dimension has a constraint.
-        x_m, _ = x.shape
-        w_n, _ = layer.weight.shape
-        output_shape = [x_m, w_n]
```
nit: remove un-related format change

pavanimajety replied:
I changed these intentionally

LucasWilkinson reviewed (approved):
LGTM, thanks! left a couple of non-critical nits

pavanimajety committed:
Introduce an env variable to configure MAX_TOKENS_PER_EXPERT

Signed-off-by: Pavani Majety <pmajety@nvidia.com>

pavanimajety commented:
I made another small change and addressed the feedback. Could you take an other look and set the auto-merge again if it looks good? @LucasWilkinson 
Thanks!

LucasWilkinson commented on vllm/model_executor/layers/fused_moe/cutlass_moe.py:
```diff
@@ -183,7 +184,8 @@ def cutlass_moe_fp8(
 FLOAT8_E4M3_MAX = torch.finfo(torch.float8_e4m3fn).max
-MAX_TOKENS_PER_EXPERT = 65536
+MAX_TOKENS_PER_EXPERT = int(
+    os.environ.get('VLLM_MODELOPT_MAX_TOKENS_PER_EXPERT', '65536'))
```
can you please add a description of what this env var does? and add it to vllm/envs please?

DarkLight1337 replied:
Oops sorry didn't see this. Can it be done in another PR?

pavanimajety replied:
yes, I'll make it a different PR - no worries :)
